File: solutions/lua/high-scores/1/high-scores.lua
local HighScores = {}

function HighScores.__call(self, v)
    HighScores.values = v
    return HighScores
end

function HighScores.scores()
    return HighScores.values
end

function HighScores.latest()
    local values = HighScores.values
    return values[#values]
end

function HighScores.personal_best()
    local values = {table.unpack(HighScores.values)}
    table.sort(values, function (a, b) return a > b end)
    return values[1]
end

function HighScores.personal_top_three()
    local values = {table.unpack(HighScores.values)}
    table.sort(values, function (a, b) return a > b end)
    return table.move(values, 1, 3, 1, {})
end

setmetatable(HighScores, HighScores)

return HighScores

File: solutions/php/high-scores/1/HighScores.php
<?php

declare(strict_types=1);

class HighScores
{
    public $scores;
    public $latest;
    public $personalBest;
    public $personalTopThree;

    public function __construct(array $scores)
    {
        $this->scores       = $scores;
        $this->latest       = end($scores);
        $this->personalBest = max($scores);
        rsort($scores);
        $this->personalTopThree = array_slice($scores, 0, 3);
    }

}
